client/components/tinymce/plugins/mentions/plugin.jsx

```diff
@@ -13,7 +13,6 @@ import React from 'react';
  */
 import Mentions from './mentions';
 import { getSelectedSite } from 'state/ui/selectors';
-import { renderWithReduxStore } from 'lib/react-helpers';
```

Comment:
This is suspicious. Seems to be used by this file!

Reply:
Indeed — especially suspicious because that's the only line codemod changed in this file. This was also the only file under `./client/components/tinymce` codemod managed to do anything about.

With many other files related to tinymce codemod simply failed and ignored them:
   - [client/my-sites/comments/controller.js](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/my-sites/comments/controller.js)
   - [client/my-sites/sharing/controller.js](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/my-sites/sharing/controller.js)
   - [client/components/tinymce/plugins/wpcom-view/plugin.js](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/wpcom-view/plugin.js)
   - [client/components/tinymce/plugins/contact-form/plugin.jsx](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/contact-form/plugin.jsx)
   - [client/components/tinymce/plugins/simple-payments/plugin.jsx](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/simple-payments/plugin.jsx)
   - [client/components/tinymce/plugins/wplink/plugin.js](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/wplink/plugin.js)
   - [client/components/tinymce/plugins/media/plugin.jsx](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/media/plugin.jsx)

Ways to go about this:
- Before removing `renderWithReduxStore`, codemod could check if it's not used anymore. => _I don't  think we need to do this, as this seems to be the only occurrence where this happened and anyway we want to get rid of `renderWithReduxStore`._
- Simply ignore `/tinymce` directory for now => _It needs a closer look and probably just manual migration to be compatible with STR. Many of those files listed above failed for different reasons so might not make sense to modify codemod to work on each individual case._

Since this is a component (or rather a bunch of plugins of the tinymce component?), I'm a bit unsure how this actually relates to single tree rendering _views_. It's not like tinymce is rendering a whole view, but just some parts inside `#primary`? That said, it's obviously [doing something with `#primary`](https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/my-sites/comments/controller.js#L46-L56).

Reply:
Looks like the codemod is ignoring the TinyMCE stuff because those aren't page.js middlewares (which is what the codemod looks for), e.g. https://github.com/Automattic/wp-calypso/blob/f1f5dd0ba751505c73bcf73a76f86f52cf651df6/client/components/tinymce/plugins/contact-form/plugin.jsx#L52

Looks like it'll require manual intervention, yeah. Shouldn't hopefully be too hard. As for details to _what exactly_ it does, I'm not sure either, but the fact that it uses `renderWithReduxStore()` is enough for us I think.